src/main/java/com/jme3/math/Quaternion.java

```diff
@@ -803,110 +803,110 @@ public float toAngleAxis(Vector3f axisStore) {
 		return angle;
 	}
 
+	public float angleBetween(Quaternion q2) {
+		float w = this.w * q2.w + this.x * q2.x + this.y * q2.y + this.z * q2.z;
+		float x = this.w * q2.x - this.x * q2.w - this.y * q2.z + this.z * q2.y;
+		float y = this.w * q2.y + this.x * q2.z - this.y * q2.w - this.z * q2.x;
+		float z = this.w * q2.z - this.x * q2.y + this.y * q2.x - this.z * q2.w;
+
+		// compute cosine and sine of the angle between
+		// do so in a numerically stable way
+		return FastMath.atan2(FastMath.sqrt(x * x + y * y + z * z), w);
+	}
+
+	public Quaternion pureSlerpLocal(Quaternion q2, float t) {
+		// make it nice and symmetrical
+		Quaternion q1 = this;
+
+		// get q2 relative to q1
+		float rw = q1.w * q2.w + q1.x * q2.x + q1.y * q2.y + q1.z * q2.z;
+		float rx = q1.w * q2.x - q1.x * q2.w - q1.y * q2.z + q1.z * q2.y;
+		float ry = q1.w * q2.y + q1.x * q2.z - q1.y * q2.w - q1.z * q2.x;
+		float rz = q1.w * q2.z - q1.x * q2.y + q1.y * q2.x - q1.z * q2.w;
+
+		// compute theta robustly
+		float theta = FastMath.atan2(FastMath.sqrt(rx * rx + ry * ry + rz * rz), rw);
+
+		// compute interpolation variables
+		float s0 = FastMath.sin((1.0f - t) * theta);
+		float s1 = FastMath.sin(t * theta);
+
+		// compute interpolated quaternion
+		float sw = s0 * q1.w + s1 * q2.w;
+		float sx = s0 * q1.x + s1 * q2.x;
+		float sy = s0 * q1.y + s1 * q2.y;
+		float sz = s0 * q1.z + s1 * q2.z;
+
+		// compute the length of the quaternion
+		float mag = FastMath.sqrt(sw * sw + sx * sx + sy * sy + sz * sz);
+
+		if (mag > 0.0f) {
+			float iMag = 1.0f / mag;
+			this.w = iMag * sw;
+			this.x = iMag * sx;
+			this.y = iMag * sy;
+			this.z = iMag * sz;
+
+		} else if (t < 0.5f) {
+			// this == q1, no need to do anything.
+		} else {
+			this.w = q2.w;
+			this.x = q2.x;
+			this.y = q2.y;
+			this.z = q2.z;
+		}
```

Comment:
Still not sure I like this, would this work as a compromise?
```suggestion
		} else if (t >= 0.5f) {
			this.w = q2.w;
			this.x = q2.x;
			this.y = q2.y;
			this.z = q2.z;
		}
		// else this == q1, no need to do anything.
```